lib/blood_contracts/core/tuple.rb

```diff
@@ -65,6 +71,18 @@ def inherited(new_klass)
 
       private
 
+      # Generate an anonimous type from a block
+      #
+      # @param Proc block which will be evaluated in a context
+      # of our anonimous type
+      #
+      # @return Class
+      #
+      def type_from_block(&block)
+        raise ArgumentError unless block_given?
+        Class.new(Refined) { class_eval(&block) }
```

Comment:
Let's give it a name, we have name of the attribute anyway. WDYT?
`const_set("InlineType_{name}", Class.new(Refined) { class_eval(&block) })`